apps/plugin-browser/src/pages/popup/popup-contents/action-center/quick-note-action.tsx

```diff
@@ -1,33 +1,125 @@
+import { extractBaseUrl, VersionedUrl } from "@blockprotocol/type-system";
 import { Button } from "@hashintel/design-system";
-import { Subgraph } from "@local/hash-subgraph";
+import { TextToken } from "@local/hash-graphql-shared/graphql/types";
+import { paragraphBlockComponentId } from "@local/hash-isomorphic-utils/blocks";
+import { Entity, EntityPropertiesObject, LinkData } from "@local/hash-subgraph";
 import { Box } from "@mui/material";
 
+import {
+  CreateEntityMutation,
+  CreateEntityMutationVariables,
+} from "../../../../graphql/api-types.gen";
+import { createEntityMutation } from "../../../../graphql/queries/entity.queries";
 import { queryApi } from "../../../../shared/query-api";
 import { useSessionStorage } from "../../../shared/use-storage-sync";
 import { Action } from "./action";
 import { QuickNoteIcon } from "./quick-note-action/quick-note-icon";
 import { TextFieldWithDarkMode } from "./text-field-with-dark-mode";
 
-const createEntityQuery = /* GraphQL */ `
-  mutation createEntity(
-    $entityTypeId: VersionedUrl!
-    $properties: EntityPropertiesObject!
-  ) {
-    createEntity(entityTypeId: $entityTypeId, properties: $properties)
-  }
-`;
-
-const createQuickNote = (text: string) => {
-  return queryApi(createEntityQuery, {
-    entityTypeId:
-      "https://app.hash.ai/@ciaran/types/entity-type/quick-note/v/1",
-    properties: {
-      "https://blockprotocol.org/@blockprotocol/types/property-type/textual-content/":
-        text,
+const quickNoteEntityTypeId =
+  "https://hash.ai/@hash/types/entity-type/quick-note/v/1" as const;
```

Comment:
@CiaranMn I ended up hard-coding these in the end because importing the systemTypes would require setting the relevant environment variables at process.env using webpack, which ended up being harder than expected and eventually didn't want to sink more time into.

Reply:
In a follow-up we can avoid hardcoding the property base urls, since we have them typed now, and can write strings for keys that are checked against the type. The entity type id doesn't exist in the generated types, although it would be helpful if it did.